test/unit/mri.controller.spec.js

```diff
@@ -258,42 +322,53 @@ describe('MRI Controller: ', function () {
     });
     
     describe('apiMriPost function() ', function() {
+
+        after(function(done) {
+            db.get('mri').remove({ filename: 'T1.mgz', source: 'https://s3.amazonaws.com/fcp-indi/data/Projects/ABIDE_Initiative/Outputs/freesurfer/5.1/CMU_a_0050642/mri/T1.mgz'});
+            done();
+        });
         it('should work correctly and make the right calls when input is correct', async function() {
             let req = {
                 db: db,
                 body: {},
                 query: {
-                    url: 'https://brainbox.pasteur.fr/mri?url=https://s3.amazonaws.com/fcp-indi/data/Projects/ABIDE_Initiative/Outputs/freesurfer/5.1/CMU_a_0050642/mri/T1.mgz',
+                    url: 'https://s3.amazonaws.com/fcp-indi/data/Projects/ABIDE_Initiative/Outputs/freesurfer/5.1/CMU_a_0050642/mri/T1.mgz',
                 },
                 user: {
                     username: ''
                 },
+                headers: {
+                    'x-forwarded-for' : 'anyone'
+                },
                 dirname: __dirname.split('/test')[0],
                 isAuthenticated: function() {
                     return this.user.username ? true : false;
                 },
                 isTokenAuthenticated: false
             };
+            console.log(req.dirname);
             let resSpy = sinon.spy();
             let jsonSpy = sinon.spy();
             let res = {
                 send: resSpy,
                 status: sinon.stub().returns({ json: jsonSpy}),
                 json: jsonSpy
             };
+            atlasMakerServer.dataDirectory = __dirname.split('/test')[0] + '/public';
             await mriController.apiMriPost(req, res);
```

Comment:
if I understand correctly, with the changes you did before, apiMriPost now will only return once the file is completely downloaded (for some very large files, this may sometimes take a few minutes!). Previously, apiMriPost would return immediately. I'm not sure if that's really the case. Could you check?

Reply:
In any case, you still need to do the tests for the polling mechanism.
• one test should check that a download is properly triggered (the server answers "downloading")
• another test should check that a download can fail: after download being properly triggered, the download would fail, and the server will answer "failed" (I don't know exactly what's the failing message)
• finally, another test should check that a download succeeds: after being properly triggered, download succeeds, file info is added to the DB, and now the server answers "succeed".
For the 2nd and the 3rd tests you will need to use setInterval(checkFunction, 2000), for checking every 2 seconds what's the server answering. You can put this setInterval call within a promise, and resolve with the final answer from the server, be it succeed or fail depending on the URL.